### PR TITLE
Export security-analysis results

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

File: src/components/results/securityanalysis/security-analysis-result-n.tsx [hunk 1]
```diff
@@ -47,6 +59,46 @@ export const SecurityAnalysisResultN: FunctionComponent<
...
+    const exportResultCsv = useCallback(() => {
+        downloadSecurityAnalysisResultZippedCsv(
+            studyUuid,
+            nodeUuid,
+            {
+                resultType: RESULT_TYPE.N,
+            },
+            csvHeaders,
+            enumValueTranslations
+        ).then((fileBlob) => downloadZipFile(fileBlob, 'n-results.zip'));
```
Comment (Meklo) [inline src/components/results/securityanalysis/security-analysis-result-n.tsx:99]: add error handling ?

File: src/components/results/securityanalysis/security-analysis-result-nmk.tsx [hunk 2]
```diff
@@ -188,6 +192,98 @@ export const SecurityAnalysisResultNmk: FunctionComponent<
...
+            studyUuid,
+            nodeUuid,
+            {
+                resultType: isFromContingency
+                    ? RESULT_TYPE.NMK_CONTINGENCIES
+                    : RESULT_TYPE.NMK_LIMIT_VIOLATIONS,
+            },
+            csvHeaders,
+            enumValueTranslations
+        ).then((fileBlob) => downloadZipFile(fileBlob, 'nmk-results.zip'));
```
Comment (Meklo) [inline src/components/results/securityanalysis/security-analysis-result-nmk.tsx:278]: add error handling here too ?

File: package.json [hunk 3]
```diff
@@ -31,8 +31,8 @@
         "geolib": "^3.2.1",
         "localized-countries": "^2.0.0",
         "lucene-escape-query": "^1.0.1",
-        "mjolnir.js": "^2.4.0",
         "mapbox-gl": "^3.0.0",
+        "mjolnir.js": "^2.4.0",
```
Comment (Meklo) [inline package.json:35]: why this move ?

File: src/components/results/securityanalysis/security-analysis-table.tsx [hunk 4]
```diff
@@ -56,13 +73,32 @@ export const SecurityAnalysisTable: FunctionComponent<
...
+                </Box>
+                <Box>
+                    <IconButton
+                        disabled={!rowsToShow || rowsToShow.length === 0}
+                        aria-label="exportCSVButton"
+                        onClick={exportCsv}
+                    >
+                        <GetAppIcon />
+                    </IconButton>
+                </Box>
```
Comment (Meklo) [inline src/components/results/securityanalysis/security-analysis-table.tsx:90]: Wouldn't it be interesting to reuse and evolve `CsvExport` ?

**Comment by Meklo** [comment]
I confirm I also have the issue with Edge v97

File: package.json (hunk 3 above)
Comment (Tristan-WorkGH) [reply package.json:35]: npm automatically sort dependencies list

File: src/components/results/securityanalysis/security-analysis-result-n.tsx (hunk 1 above)
Comment (klesaulnier) [reply src/components/results/securityanalysis/security-analysis-result-n.tsx:99]: fone

File: src/components/results/securityanalysis/security-analysis-result-nmk.tsx (hunk 2 above)
Comment (klesaulnier) [reply src/components/results/securityanalysis/security-analysis-result-nmk.tsx:278]: done

File: src/components/results/securityanalysis/security-analysis-table.tsx (hunk 4 above)
Comment (klesaulnier) [reply src/components/results/securityanalysis/security-analysis-table.tsx:90]: done for the visual aspect
They're not doing the same thing technically speaking though, one is using AGGrid to create CSV from frontend, the other is fully fetching the csv from the back

File: src/components/results/securityanalysis/security-analysis-result-n.tsx [hunk 5]
```diff
@@ -58,11 +72,38 @@ export const SecurityAnalysisResultN: FunctionComponent<
         [intl, sortProps, filterProps, filterEnums]
     );
 
+    const csvHeaders = useMemo(
+        () => columnDefs.map((cDef) => cDef.headerName),
+        [columnDefs]
+    );
```
Comment (Tristan-WorkGH) [inline src/components/results/securityanalysis/security-analysis-result-n.tsx:78]: useMemo from useMemo?  
maybe doing it as one call?
```suggestion
    const [csvHeaders, columnDefs] = useMemo(() => {
        let columnDefs = securityAnalysisTableNColumnsDefinition(intl, sortProps, filterProps, filterEnums);
        return [columnDefs, columnDefs.map((cDef) => cDef.headerName)];
    }, [intl, sortProps, filterProps, filterEnums]);
```

File: src/components/results/securityanalysis/security-analysis-result-nmk.tsx [hunk 6]
```diff
@@ -188,6 +192,40 @@ export const SecurityAnalysisResultNmk: FunctionComponent<
         [isFromContingency, theme.selectedRow.background]
     );
 
+    const csvHeaders = useMemo(
+        () => columnDefs.map((cDef) => cDef.headerName),
+        [columnDefs]
+    );
```
Comment (Tristan-WorkGH) [inline src/components/results/securityanalysis/security-analysis-result-nmk.tsx:198]: same

File: src/components/results/securityanalysis/security-analysis-table.tsx [hunk 7]
```diff
@@ -56,13 +74,24 @@ export const SecurityAnalysisTable: FunctionComponent<
...
-            defaultColDef={defaultColDef}
-            onGridReady={onGridReady}
-            overlayNoRowsTemplate={overlayNoRowsTemplate}
-            {...agGridProps}
-        />
+        <Box sx={styles.gridContainer}>
+            <Box sx={styles.csvExport}>
+                <Box style={{ flexGrow: 1 }}></Box>
+                <ExportButton
+                    disabled={!rowsToShow || rowsToShow.length === 0}
```
Comment (Tristan-WorkGH) [inline src/components/results/securityanalysis/security-analysis-table.tsx:81]: would it work to do
```suggestion
                    disabled={rowsToShow?.length <= 0}
```
?

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

**Review by Meklo — APPROVED** [review]

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**